File: BondPrototype/Angular/src/app/api/models/api-models/person-result.ts



export class PersonResult {
    public id: number;
    public name: string;
    public dateOfBirth: Date | null;
    public picture: string;
    public biography: string;
    public directed: PersonResult.DirectedOrActedInMovie[];
    public actedIn: PersonResult.DirectedOrActedInMovie[];
}

export namespace PersonResult {
    export class DirectedOrActedInMovie {
        constructor(public id: number, public title: string, public rating: number, public releaseDate: Date) {
        }
    }
}

File: BondPrototype/Models/ApiModels/PersonResult.cs
namespace BondPrototype.Models;

public class PersonResult
{
    public int Id { get; set; }
    public string Name { get; set; }
    public DateTime? DateOfBirth { get; set; }
    public string Picture { get; set; }
    public string Biography { get; set; }

    public record struct DirectedOrActedInMovie(int Id, string Title, byte Rating, DateTime ReleaseDate);

    public List<DirectedOrActedInMovie> Directed { get; set; }
    public List<DirectedOrActedInMovie> ActedIn { get; set; }
}